tests/javascripts/searchActivity.test.js

```diff
@@ -0,0 +1,189 @@
+                    <td class="table-field time-sent">2024-08-20 10:00</td>
+                    <td class="table-field sender">Sender 1</td>
+                    <td class="table-field report"><a href="#"><img src="#" alt="File Download Icon"></a></td>
+                    <td class="table-field delivered">10</td>
+                    <td class="table-field failed">2</td>
+                </tr>
+                <tr class="table-row" style="display: table-row;">
+                    <td class="table-field jobid"><a href="#">Job 2</a></td>
+                    <td class="table-field template">Template 2</td>
+                    <td class="table-field time-sent">2024-08-20 11:00</td>
+                    <td class="table-field sender">Sender 2</td>
+                    <td class="table-field report"><a href="#"><img src="#" alt="File Download Icon"></a></td>
+                    <td class="table-field delivered">5</td>
+                    <td class="table-field failed">0</td>
+                </tr>
+                <tr class="table-row" style="display: table-row;">
+                    <td class="table-field jobid"><a href="#">Job 3</a></td>
+                    <td class="table-field template">Template 3</td>
+                    <td class="table-field time-sent">2024-08-20 12:00</td>
+                    <td class="table-field sender">Sender 3</td>
+                    <td class="table-field report"><a href="#"><img src="#" alt="File Download Icon"></a></td>
+                    <td class="table-field delivered">0</td>
+                    <td class="table-field failed">1</td>
+                </tr>
+            </tbody>
+        </table>
+        <div class="usa-table__announcement-region usa-sr-only" aria-live="polite" aria-atomic="true">
+            <!-- Live region content goes here -->
+        </div>
+    </div>
+    `;
+
+    // Attach the filterTable function to the window object for testing
+    window.filterTable = function() {
+        const searchInput = document.getElementById('all-activity-search');
+        const tableRows = document.querySelectorAll('.job-table tbody .table-row');
+        const tableBody = document.querySelector('.job-table tbody');
+        const liveRegion = document.querySelector('.usa-table__announcement-region');
+
+        const filter = searchInput.value.toLowerCase();
+        let visibleRowCount = 0;
+
+        tableRows.forEach(row => {
+            const rowText = row.textContent.toLowerCase();
+
+            if (rowText.includes(filter)) {
+                row.style.display = 'table-row'; // Show the row
+                visibleRowCount++;
+            } else {
+                row.style.display = 'none'; // Hide the row
+            }
+        });
+
+        // Update ARIA live region
+        liveRegion.textContent = `${visibleRowCount} result${visibleRowCount !== 1 ? 's' : ''} found`;
+
+        // Check if there are no visible rows
+        if (visibleRowCount === 0) {
```

Comment:
If this is the same code that is in searchActivity regarding business logic, it might make sense to extract this logic to a generic helper file so that we can reuse that same logic in the javascript tests. This way if future updates are needed, it only needs to be done in one place instead of multiple. 

I am still learning context and might be missing something here, so ill go ahead and approve, but if you think its a good idea as well then you can make that change!

Reply:
@alexjanousekGSA - we're actually waiting on some back end work to provide an endpoint to search against, so this might change. When that happens I'll reach back out to see if it's worth editing for testing-sake.